### PR TITLE
Implements basic site import

### PR DESCRIPTION
**Pull request by cperryk**
Related: #19. Implements basic site import. Does not implement lists or users import.

### CODE REVIEW

File: lib/cmd/import.js [hunk 1]
```diff
@@ -128,15 +126,39 @@ function importSingleUrl(url, prefix, argv) {
...
-  logger.error('Site import not supported yet!');
+/**
+ * import data from a site
+ * @param {string} site
+ * @param {string} prefix of site to import into
+ * @param {object} argv
+ * @return {Stream}
+ */
+function importSite(site, prefix, argv) {
+  const pagesEndpoint = site + '/pages',
```
Comment (nelsonpecora) [inline lib/cmd/import.js]: Ah, this offset won't really work, because the page list doesn't really have ordering. (It's deterministic based on uid, but those uids have different schemes based on where they have been generated, e.g. amphora vs kiln vs scratch importer vs claycli vs bootstrap). We should discuss this in person (and I apologize, because I thought I had written out the plan for this somewhere but I can't find it)

The general idea was to query the built-in `pages` elastic index, since that'll have all of the pages in a sortable order (as well as the published status), then base our offset/size off of that.

File: lib/cmd/import.js [hunk 2]
```diff
@@ -109,8 +109,6 @@ function importStream(prefix, argv) {
     // map agnostic chunks to data we can save
     .map(mapChunksToData(prefix))
     .flatMap((item) => _.includes(item.url, '/uris') ? rest.put(item, key, argv.concurrency, 'text') : rest.put(item, key, argv.concurrency))
-    .map(showProgress(argv))
-    .toArray(showCompleted);
```
Comment (nelsonpecora) [inline lib/cmd/import.js]: Why were these removed...?

File: lib/cmd/import.js [hunk 3]
```diff
@@ -128,15 +126,39 @@ function importSingleUrl(url, prefix, argv) {
   return clayInput.importUrl(url, argv.concurrency)
     .flatMap(chunks.replacePrefixes(prefix))
     .stopOnError(fatalError) // exit early if there's a problem reaching the input clay instance
-    .flatMap((item) => rest.put(item, key, argv.concurrency))
-    .map(showProgress(argv))
-    .toArray(showCompleted);
```
Comment (nelsonpecora) [inline lib/cmd/import.js]: Why were these removed?

File: lib/cmd/import.js [hunk 4]
```diff
@@ -208,19 +229,23 @@ function handler(argv) {
...
-    return importFile(config.getFile(argv.file), prefix, argv);
+    stream = importFile(config.getFile(argv.file), prefix, argv);
   } else {
     logger.error('Please specify somewhere to import from!');
     process.exit(1);
   }
+
+  stream
+    .map(showProgress(argv))
+    .toArray(showCompleted);
```
Comment (nelsonpecora) [inline lib/cmd/import.js]: Oh. I guess this makes sense, then.

**Comment by cperryk** [comment]
See https://github.com/clay/claycli/issues/19 for notes on planned changes

**Comment by cperryk** [comment]
Closing in favor of #37

**Comment by cperryk** [comment]
Closing in favor of #37
